File: 1.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        order = sorted(range(len(nums)), key=nums.__getitem__)
        nums.sort()
        i = 0
        j = len(nums) - 1
        while True:
            current = nums[i] + nums[j]
            if current < target:
                i += 1
            elif current > target:
                j -= 1
            else:
                return [order[i], order[j]]

File: 1.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        for i in 0..nums.len() {
            for j in 0..i {
                if nums[i] + nums[j] == target {
                    return vec![i as i32, j as i32]
                }
            }
        }
        panic!();
    }
}
